### PR TITLE
Access ctx in resolve

### LINKED ISSUE #12: Access Hono context in the resolver
In current design, 
```ts
type Options = {
  schema: GraphQLSchema
  rootValue?: unknown
  pretty?: boolean
  validationRules?: ReadonlyArray<ValidationRule>
  // graphiql?: boolean
}

export const graphqlServer = (options: Options) => {
...
   return async (c: Context, next: Next) => {
      const result = await execute({
            schema,
            document: documentAST,
            rootValue,
            variableValues: variables,
            operationName: operationName,
      })
     ctx.json(result)
  }
...

}
```
`rootValue` is part of options of the middleware, the Hono `ctx` is not accessible for rootValue (resolvers), it's not very convenient for some usage, like this one https://github.com/honojs/hono/issues/278, so I am thinking to make the `Options.rootValue` to be a function like,

```ts
export type RootValue = (ctx: Context) => any

export const graphqlServer = (options: Options) => {
...
   return async (c: Context, next: Next) => {
      const result = await execute({
            schema,
            document: documentAST,
            rootValue: rootValue(ctx),
            variableValues: variables,
            operationName: operationName,
      })
     ctx.json(result)
  }
...

}
```

### PR DESCRIPTION
**Pull request by metrue**
Close #12

### CODE REVIEW

File: src/index.test.ts [hunk 1]
```diff
@@ -26,34 +28,32 @@ describe('errorMessages', () => {
...
-    graphqlServer({
-      schema,
-      rootValue,
-    })
-  )
+    app.use(
+      '/graphql',
+      graphqlServer({
+        schema,
+        rootResovler: () => rootValue,
```
Comment (yusukebe) [inline src/index.test.ts:48]: Is it a typo?

Comment (metrue) [reply src/index.test.ts:48]: @yusukebe 
No, it's not a typo, I am thinking to make `ctx` to be accessible in resolver,  so I am trying to change the `option.rootValue` to be function, but I am still in the thinking, the detail idea is described here, https://github.com/honojs/graphql-server/issues/12, what do you guys think? cc @usualoma

Comment (yusukebe) [reply src/index.test.ts:48]: I think `rootResovler` should be `rootResolver` 🤔

Comment (metrue) [reply src/index.test.ts:48]: Oh, my mistake.

**Review by yusukebe — APPROVED** [review]
Looks good!

**Pull request merged**